src/thread.rs

```diff
@@ -104,10 +114,182 @@ enum ThreadJoinStatus {
     Joined,
 }
 
+/// Set of sync objects that can have properties queried.
+/// The futex is not included since it can only signal
+/// and awake values.
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
+enum SyncObject {
```

Comment:
The primary purpose of this seems to be "object that can be watched by a thread", so that should be the first thing the doc comment says.

The part about "having properties queried" seems to be important, but I do not understand it.